flocker/node/test/test_loop.py

```diff
@@ -430,6 +435,55 @@ def test_convergence_sent_state_fail_resends(self):
             )
         )
 
+    def test_convergence_sent_state_fail_resends_alternating(self):
+        """
+        If sending state to the control node fails the next iteration will send
+        state even if the state is the same as the last acknowledge state.
+        """
+        local_state = NodeState(
+            hostname=u'192.0.2.123',
+            used_ports=pset(), applications=pset(),
+        )
+        changed_local_state = local_state.set(
+            used_ports=pset([80]),
+        )
+        configuration = Deployment(nodes=[to_node(local_state)])
+        state = DeploymentState(nodes=[local_state])
+        changed_state = DeploymentState(nodes=[changed_local_state])
+        deployer = ControllableDeployer(
+            local_state.hostname,
+            [succeed(local_state), succeed(changed_local_state),
+             succeed(local_state)],
+            [no_action(), no_action(), no_action()])
+        client = self.make_amp_client(
+            [local_state, changed_local_state.copy()],
+            successes=[True, False],
```

Comment:
:question: This is another key part of this test, right? The second AMP request fails. Maybe worth a comment.